src/rules/quotemarkRule.ts

```diff
@@ -67,13 +71,15 @@ class QuoteWalker extends Lint.RuleWalker {
             const firstCharacter = text.charAt(0);
             const lastCharacter = text.charAt(text.length - 1);
 
-            if (this.quoteMark === QuoteMark.SINGLE_QUOTES) {
-                if (firstCharacter !== "'" || lastCharacter !== "'") {
-                    failure = this.createFailure(position, width, Rule.SINGLE_QUOTE_FAILURE);
-                }
-            } else if (this.quoteMark === QuoteMark.DOUBLE_QUOTES) {
-                if (firstCharacter !== "\"" || lastCharacter !== "\"") {
-                    failure = this.createFailure(position, width, Rule.DOUBLE_QUOTE_FAILURE);
+            const expectedQuoteMarkCharacter = getQuoteMarkCharacter(this.quoteMark);
+
+            if (firstCharacter !== expectedQuoteMarkCharacter || lastCharacter !== expectedQuoteMarkCharacter) {
+                // allow the "other" quote mark to be used, but only to avoid having to escape
+                const isIncludingQuoteMark = text.slice(1, -1).indexOf(expectedQuoteMarkCharacter) !== -1;
+
+                if (!(this.allowEscape && isIncludingQuoteMark)) {
+                    const failureMessage = getFailureMessage(this.quoteMark);
```

Comment:
Same reason of the [comment](https://github.com/palantir/tslint/pull/611#discussion_r38621628).